include/device.h

```diff
@@ -497,6 +497,47 @@ device_required_handles_get(const struct device *dev,
 	return rv;
 }
 
+/**
+ * @brief Get the set of handles that this device supports.
+ *
+ * The set of supported devices is inferred from devicetree, and does not
+ * include any software constructs that may depend on the device.
```

Comment:
Looking at this again a second time, is there any reason for the asymmetry here?

We have dependencies both from devicetree and declared by the build system as "injected" dependencies. So if a device D1 which is not created from devicetree requires device D2 that is created from devicetree, we still know that D2 supports D1 at gen_handles.py time. Why not include it here?

Reply:
This comment is perhaps a little bit misleading for the supports direction, as a software construct CANNOT inject any dependencies. This is because `DEVICE_DEFINE` does not expose the required arguments to `Z_DEVICE_DEFINE`.

To be honest I'm not entirely sure of the purpose of the injected dependencies in the current implementation. Any ordinals passed to the constructor are not transformed by `gen_handles.py`, and ordinals are generally useless. If a dependency has an ordinal it implies that it is from devicetree, so why is the dependency not expressed in devicetree in the first place? As already noted, a software device cannot express dependencies.